packages/core/src/hooks/use-iso-layout-effect.ts

```diff
@@ -0,0 +1,7 @@
+'use client';
+
+import { useLayoutEffect } from 'react';
+
+const useIsoLayoutEffect = globalThis?.document ? useLayoutEffect : () => {};
+
+export { useIsoLayoutEffect };
```

Comment:
useIsomorphicEffect is not recommended because of [rules of react](https://smoores.dev/post/no_such_thing_isomorphic_layout_effect/).

It is better to use like [react-aria's approach](https://github.com/adobe/react-spectrum/blob/7a8c11aa37b7cf20d17e429bd8122310bc0cb32d/packages/%40react-aria/ssr/src/SSRProvider.tsx#L145-L168) / ([2](https://github.com/adobe/react-spectrum/blob/7a8c11aa37b7cf20d17e429bd8122310bc0cb32d/packages/%40react-aria/utils/src/useId.ts#L40-L84)).